### DISCUSSION TITLE
Username and Password not working?

### DISCUSSION THREAD

**Discussion by OaShadow** [bluenviron/mediamtx#1230] [Questions] [answered]
Hello there,
so my question is, is the username and password feature broken or am I just not smart enought to use it.
I was trying to insert credentials like "test" for both to test the credentials feature.
Now every time I try to connect the INF in the CLI just shows me my client got "invalid credentials"

I was also trying to use the sha256 key from test (sha256:9f86d081884c7d659a2feaa0c55ad015a3bf4f1b2b0b822cd15d6c15b0f00a08) but it also says invalid credentials.
The IP I'am trying to connect to is build like this: _**rtmp://test:test@123.123.123.123:1935/stream/path**_

I hope the information is enought to help me out. 

Thanks.

**Comment by aler9** [comment, marked as answer]
in order to provide credentials with the RTMP protocol, you have to append them to the URL in this way, as explained in [the readme](https://github.com/aler9/rtsp-simple-server#general-usage-1):

rtmp://123.123.123.123:1935/stream/path?user=test&pass=test

Unfortunately credentials passed through the authentication section of the URL can't be supported due to a limitation of the RTMP protocol, since these are passed by the client to the server before the client tells the server whether it is publishing or is reading, and therefore we can't use separate credentials for readers and publishers.

i.e:

```
client -> server: connect to path /test
server -> client: please provide credentials
client -> server: here are the credentials
server -> client: ok
client -> server: i want to publish / read
```

**Reply by OaShadow** [reply]
It is working with the URL-Part "?user=test&pass=test", thank you ^^

[a comment by an automated account is not shown]
